static/js/recent_topics_ui.js

```diff
@@ -707,6 +707,12 @@ function is_focus_at_last_table_row() {
     return row_focus === $topic_rows.length - 1;
 }
 
+function has_unread(row) {
+    const $topic_rows = $("#recent_topics_table table tbody tr");
+    const $topic_row = $topic_rows.eq(row);
+    return !$topic_row.find(".recent_topic_name span").hasClass("unread_hidden");
+}
```

Comment:
Instead of using DOM for this, we can just check if the topic is unread or not using what we already know about it. Using DOM will take a lot of time and resources.

Reply:
I have updated the code. For getting rows i have used jQuery but to check if it is unread, i have used `unread.js`. Is there any other ways to get rows without jQuery ?